kmip/pie/client.py

```diff
@@ -0,0 +1,340 @@
+            AttributeType.CRYPTOGRAPHIC_USAGE_MASK,
+            key.cryptographic_usage_masks)
+
+        attributes = [mask_attribute]
+        template = TemplateAttribute(attributes=attributes)
+
+        object_type = key.object_type
+
+        # Register the secret and handle the results
+        secret = self.object_factory.convert(key)
+        result = self.proxy.register(object_type, template, secret)
+
+        status = result.result_status.enum
+        if status == ResultStatus.SUCCESS:
+            uid = result.uuid.value
+            return uid
+        else:
+            reason = result.result_reason.enum
+            message = result.result_message.value
+            raise KmipOperationFailure(status, reason, message)
+
+    def get_secret(self, uid):
+        """
+        Get a secret from a KMIP appliance.
+
+        Args:
+            uid (string): The unique ID of the secret to retrieve.
+
+        Raises:
+            ClientConnectionNotOpen: if the client connection is unusable
+            KmipOperationFailure: if the operation result is a failure
+            TypeError: if the input argument is invalid
+        """
+        # Check input
+        if not isinstance(uid, six.string_types):
+            raise TypeError("uid must be a string")
+
+        # Verify that operations can be given at this time
+        if not self._is_open:
+            raise ClientConnectionNotOpen()
+
+        # Get the secret and handle the results
+        result = self.proxy.get(uid)
+
+        status = result.result_status.enum
+        if status == ResultStatus.SUCCESS:
+            secret = self.object_factory.convert(result.secret)
+            return secret
+        else:
+            reason = result.result_reason.enum
+            message = result.result_message.value
+            raise KmipOperationFailure(status, reason, message)
+
+    def destroy_secret(self, uid):
+        """
+        Destroy a secret stored by a KMIP appliance.
+
+        Args:
+            uid (string): The unique ID of the secret to destroy.
+
+        Raises:
+            ClientConnectionNotOpen: if the client connection is unusable
+            KmipOperationFailure: if the operation result is a failure
+            TypeError: if the input argument is invalid
+        """
+        # Check input
+        if not isinstance(uid, six.string_types):
+            raise TypeError("uid must be a string")
+
+        # Verify that operations can be given at this time
+        if not self._is_open:
+            raise ClientConnectionNotOpen()
+
+        # Destroy the secret and handle the results
+        result = self.proxy.destroy(uid)
+
+        status = result.result_status.enum
+        if status == ResultStatus.SUCCESS:
+            return
+        else:
+            reason = result.result_reason.enum
+            message = result.result_message.value
+            raise KmipOperationFailure(status, reason, message)
+
+    def __enter__(self):
```

Comment:
They allow you to use the client as a context manager. You can now do this: `with KmipClient(...) as client:`. It automatically opens and closes the client for you.